### PR TITLE
SECURITY: make conduct section, warn against weaponized PRs

### PR DESCRIPTION
**Pull request by woodruffw**
Per Slack discussion: researchers are sometimes tempted to submit public PRs containing pentesting/partially proven-out exploits. We should discourage that explicitly.

(Language is not settled here, please aggressively nitpick!)

CC @p-linnane @Bo98

### CODE REVIEW

File: SECURITY.md [hunk 1]
```diff
@@ -39,10 +39,12 @@ The following do not constitute security vulnerabilities in Homebrew:
...
+## Conduct
+
+While researching, we ask you to refrain from:
 
 - Denial of service
 - Spamming
 - Social engineering (including phishing) of Homebrew maintainers or contributors
 - Any physical attempts against Homebrew's machines
-
+- Testing discoveries on Homebrew's CI/CD or other services by filing public PRs containing weaknesses
```
Comment (Bo98) [inline SECURITY.md:50]: Can maybe make it more general such as

```suggestion
- Performing vulnerability research in public, such as testing discoveries by opening pull requests to Homebrew's public repositories without prior written approval
```

(you can reword this further)

**Review by Bo98 — APPROVED** [review]
Ah nice! Didn't realise there were some conduct rules here before - good it's much more obvious now with its own heading

File: SECURITY.md (hunk 1 above)
Comment (woodruffw) [reply SECURITY.md:50]: I like your wording!

**Review by SMillerDev — APPROVED** [review]

**Review by issyl0 — APPROVED** [review]
Thanks!

**Review by MikeMcQuaid — APPROVED** [review]
Great addition, thanks!

**Pull request merged**